src/quo2/components/drawers/action_drawers.cljs

```diff
@@ -9,36 +9,40 @@
     colors/danger-50
     (colors/theme-colors colors/neutral-50 colors/neutral-40)))
 
-(defn divider []
+(defn divider [accessibility-label]
   [rn/view {:style {:border-top-width 1
                     :border-top-color (colors/theme-colors
                                        colors/neutral-10 colors/neutral-90)
                     :margin-top       8
                     :margin-bottom    7
                     :align-items      :center
-                    :flex-direction   :row}}])
+                    :flex-direction   :row}
+            :accessible true
+            :test-ID (str "divider-" accessibility-label)}])
 
 (defn action [{:keys [icon
                       label
                       sub-label
                       right-icon
                       danger?
                       on-press
-                      add-divider?] :as action-props}]
+                      add-divider?
+                      accessibility-label] :as action-props}]
   (when action-props
     [:<> {:key label}
-     (when add-divider? [divider])
-     [rn/touchable-highlight {:style          {:border-radius     12
+     (when add-divider? [divider accessibility-label])
+     [rn/touchable-highlight {:accessibility-label accessibility-label
+                              :style          {:border-radius     12
                                                :height (if sub-label 58 50)
                                                :margin-horizontal 8}
                               :underlay-color (colors/theme-colors colors/neutral-5 colors/neutral-90)
                               :on-press       on-press}
       [rn/view {:style
                 {:height (if sub-label 58 50)
-
                  :margin-horizontal 12
                  :flex-direction    :row}}
-       [rn/view {:style
+       [rn/view {:test-ID (str "icon-" accessibility-label)
```

Comment:
I would argue it's generally better to avoid dynamic test IDs. We lose the ability to search for them, which is very useful once the test suit grows in size. Of course there are exceptions, but in the `action-drawers` tests we see all of them test only a single section. But even with multiple sections, it's trivial to use something like `getAllByTestId` and match against the collection.

So unless there's no way to avoid dynamic test IDs, I think we should avoid them. Well, that's just my experience using the react testing library.

Reply:
Yeah I think you are right about this. React Native Testing library says it's best to use TestID's as a last option too. I will try your solution, seems like a good option! :)